protocols/imap/src/main/java/org/apache/james/imap/api/ImapConfiguration.java

```diff
@@ -19,29 +19,57 @@
 
 package org.apache.james.imap.api;
 
+import java.util.Locale;
 import java.util.Set;
 import java.util.concurrent.TimeUnit;
 
 import org.apache.james.imap.processor.IdleProcessor;
 
+import org.apache.commons.lang.StringUtils;
+
+import com.google.common.base.Function;
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Predicate;
+import com.google.common.collect.FluentIterable;
 import com.google.common.collect.ImmutableSet;
 
 public class ImapConfiguration {
     public static Builder builder() {
         return new Builder();
     }
+
     public static class Builder {
-        private long idleTimeInterval;
+        private static final Function<String, String> NORMALIZE_STRING = new Function<String, String>() {
+            @Override
+            public String apply(String disableCap) {
+                return StringUtils.normalizeSpace(disableCap);
+            }
+        };
+        private static final Predicate<String> NO_BLANK = new Predicate<String>() {
+            @Override
+            public boolean apply(String disableCap) {
+                return noBlankOrNullString(disableCap);
+            }
+        };
+        private static final String NULL = "null";
+
+        private static boolean noBlankOrNullString(String disableCap) {
+            return !StringUtils.isBlank(disableCap) && !NULL.equalsIgnoreCase(disableCap.toLowerCase(Locale.US));
+        }
+
+        private Optional<Long> idleTimeInterval;
         private TimeUnit idleTimeIntervalUnit;
```

Comment:
This is the old behavior and if we have default value for idleTime then we should have default value for timeUnit I think.